### DISCUSSION TITLE
Parsing a stream

### DISCUSSION THREAD

**Discussion by awolff64** [danielaparker/jsoncons#381] [Q&A]
Try to parse json data that i receive in chunks. Tried to std::stringstream and then parse.
However this did not work if the parse did not get a string for a complete JSON value.

I am after something where I can push 
1. my chunks of data in
2. pull out JSON values until there is an error or not enough data.

I want to avoid collecting the input data and splitting it into strings that represent a JSON value.

Can anyone give me some advice how to do it?

**Comment by danielaparker** [comment]
Have your looked at [this example](https://github.com/danielaparker/jsoncons/blob/master/doc/ref/json_parser.md#user-content-incremental-parsing) of incremental parsing?

**Comment by awolff64** [comment]
Thanks for the answer.
Yes, I tried that.
What I am looking for is a solution that would continue parsing once one value has been successfully read.
In the second example there I am after a solution that would parse the JSON text "[10.5]{}" and return
1. array
2. object

Used a reset on the parser once a JSON value has been returned. This keeps the position in the current input. However I am not sure if this is part of the official API.

Thanks for your help.

Regards,
Andreas

**Comment by danielaparker** [comment]
Yes, `reset` is intended for that purpose.
